apps/estuary/lib/estuary/init_server.ex

```diff
@@ -0,0 +1,43 @@
+defmodule Estuary.InitServer do
+  @moduledoc """
+  Task to initialize estuary and start ingesting each previously recorded dataset
+  """
+  use GenServer
+
+  alias Estuary.DataReader
+  alias Estuary.Datasets.DatasetSchema
+  alias Estuary.DataWriter
+
+  def start_link(opts) do
+    name = Keyword.get(opts, :name, __MODULE__)
+    GenServer.start_link(__MODULE__, [], name: name)
+  end
+
+  def init(_) do
+    with :ok <- init_writer(),
+         :ok <- DataReader.init(),
+         pid when is_pid(pid) <- Process.whereis(Pipeline.DynamicSupervisor) do
+      Process.monitor(pid)
+      {:ok, %{pipeline: pid}}
+    else
+      _ -> raise "Could not initialize Estuary"
+    end
+  end
+
+  def handle_info({:DOWN, _, _, pid, _} = msg, %{pipeline: pid} = state) do
+    with :ok <- DataReader.init(),
+         pid when is_pid(pid) <- Process.whereis(Pipeline.DynamicSupervisor) do
+      Process.monitor(pid)
+      {:noreply, %{pipeline: pid}}
+    else
+      _ ->
+        Process.sleep(100)
```

Comment:
I'm not a fan of how we did this, but it technically passes the tests.
I could see us hanging here indefinitly retrying to connect to pipeline, unless another part of the app/otp says we've tried for too long. Would like to talk with Jeff or someone about this.